src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationCapability.cs

```diff
@@ -140,10 +143,13 @@ internal struct DelegationCapability
         public static readonly BigInteger AsType = BigInteger.Pow(2, 44);                       //    0x100000000000
         public static readonly BigInteger ArrayLookup = BigInteger.Pow(2, 45);                  //    0x200000000000
         public static readonly BigInteger Distinct = BigInteger.Pow(2, 46);                     //    0x400000000000
-        public static readonly BigInteger Join = BigInteger.Pow(2, 47);                         //    0x800000000000
+        public static readonly BigInteger JoinInner = BigInteger.Pow(2, 47);                    //    0x800000000000
+        public static readonly BigInteger JoinLeft = BigInteger.Pow(2, 48);                     //   0x1000000000000
+        public static readonly BigInteger JoinRight = BigInteger.Pow(2, 49);                    //   0x2000000000000
+        public static readonly BigInteger JoinFull = BigInteger.Pow(2, 50);                     //   0x4000000000000
 
         // Please update it as max value changes.
-        private static BigInteger maxSingleCapabilityValue = Join;
+        private static BigInteger maxSingleCapabilityValue = JoinFull;
 
         // Indicates support all functionality.
         public static BigInteger SupportsAll
```

Comment:
This function seems highly suspicious. 
Does adding new capabilities cause this behavior to change?

Reply:
this constant is only used for testing the validity of the capability or in tests
It's not used in DV or PA